tests/Unit/Bridge/Symfony/DependencyInjection/SearchExtensionTest.php

```diff
@@ -0,0 +1,131 @@
+use LoyaltyCorp\Search\Bridge\Symfony\Console\Commands\SearchIndexCreateCommand;
+use LoyaltyCorp\Search\Bridge\Symfony\Console\Commands\SearchIndexFillCommand;
+use LoyaltyCorp\Search\Bridge\Symfony\Console\Commands\SearchIndexLiveCommand;
+use LoyaltyCorp\Search\Bridge\Symfony\DependencyInjection\SearchExtension;
+use LoyaltyCorp\Search\Bridge\Symfony\Factories\ClientFactory;
+use LoyaltyCorp\Search\Bridge\Symfony\Factories\RegisteredSearchHandlersFactory;
+use LoyaltyCorp\Search\Bridge\Symfony\Interfaces\ClientFactoryInterface;
+use LoyaltyCorp\Search\Bridge\Symfony\Interfaces\RegisteredSearchHandlersFactoryInterface;
+use LoyaltyCorp\Search\Bridge\Symfony\Listeners\BatchOfUpdatesListener;
+use LoyaltyCorp\Search\Bridge\Symfony\Listeners\EntityUpdateListener;
+use LoyaltyCorp\Search\Client;
+use LoyaltyCorp\Search\Helpers\ClientBulkResponseHelper;
+use LoyaltyCorp\Search\Helpers\RegisteredSearchHandlers;
+use LoyaltyCorp\Search\Indexer;
+use LoyaltyCorp\Search\Indexer\AccessTokenMappingHelper;
+use LoyaltyCorp\Search\Interfaces\Access\AccessPopulatorInterface;
+use LoyaltyCorp\Search\Interfaces\ClientInterface;
+use LoyaltyCorp\Search\Interfaces\Helpers\ClientBulkResponseHelperInterface;
+use LoyaltyCorp\Search\Interfaces\Helpers\RegisteredSearchHandlersInterface;
+use LoyaltyCorp\Search\Interfaces\Indexer\MappingHelperInterface;
+use LoyaltyCorp\Search\Interfaces\IndexerInterface;
+use LoyaltyCorp\Search\Interfaces\PopulatorInterface;
+use LoyaltyCorp\Search\Interfaces\RequestProxyFactoryInterface;
+use LoyaltyCorp\Search\Interfaces\ResponseFactoryInterface;
+use LoyaltyCorp\Search\Interfaces\Transformers\IndexNameTransformerInterface;
+use LoyaltyCorp\Search\Interfaces\UpdateProcessorInterface;
+use LoyaltyCorp\Search\Interfaces\Workers\EntityUpdateWorkerInterface;
+use LoyaltyCorp\Search\Populator;
+use LoyaltyCorp\Search\RequestProxyFactory;
+use LoyaltyCorp\Search\ResponseFactory;
+use LoyaltyCorp\Search\Transformers\DefaultIndexNameTransformer;
+use LoyaltyCorp\Search\UpdateProcessor;
+use LoyaltyCorp\Search\Workers\EntityUpdateWorker;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Tests\LoyaltyCorp\Search\TestCases\UnitTestCase;
+
+/**
+ * @covers \LoyaltyCorp\Search\Bridge\Symfony\DependencyInjection\SearchExtension
+ */
+final class SearchExtensionTest extends UnitTestCase
+{
+    /**
+     * Test commands are loaded.
+     *
+     * @return void
+     *
+     * @throws \Exception
+     */
+    public function testCommandsAreLoaded(): void
+    {
+        $container = new ContainerBuilder();
+
+        (new SearchExtension())->load(['search' => ['use_commands' => true]], $container);
+
+        $commands = [
+            SearchIndexCleanCommand::class,
+            SearchIndexCreateCommand::class,
+            SearchIndexFillCommand::class,
+            SearchIndexLiveCommand::class,
+        ];
+
+        foreach ($commands as $command) {
+            self::assertTrue($container->hasDefinition($command));
```

Comment:
This should probably have a stronger assertion that the definition is tagged and returns the correct service class?

Reply:
(Same for all 3 tests?)